### PR TITLE
Add imagestreamtag restore plugin

### PR DESCRIPTION
**Pull request by sseago**

### CODE REVIEW

File: velero-plugins/common/restore.go [hunk 1]
```diff
@@ -25,6 +25,8 @@ func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*v
 	if err != nil {
 		return nil, err
 	}
+	name := metadata.GetName()
+	p.Log.Infof("common restore plugin for %s, %f", name, annotations)
```
Comment (dymurray) [inline velero-plugins/common/restore.go]: `%f` doesn't seem right. Isn't that for decimals?

File: velero-plugins/imagestream/backup.go [hunk 2]
```diff
@@ -30,12 +30,12 @@ func (p *BackupPlugin) AppliesTo() (velero.ResourceSelector, error) {
 
 // Execute copies local registry images into migration registry
 func (p *BackupPlugin) Execute(item runtime.Unstructured, backup *v1.Backup) (runtime.Unstructured, []velero.ResourceIdentifier, error) {
-	p.Log.Info("Hello from Imagestream backup plugin!!")
+	p.Log.Info("[is-backup] Hello from Imagestream backup plugin!!")
```
Comment (dymurray) [inline velero-plugins/imagestream/backup.go:33]: Good idea prefixing the log statements

File: velero-plugins/imagestreamtag/restore.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+			p.Log.Info(fmt.Sprintf("[istag-restore] Looking up reference tag: %s/%s", refNamespace, imageStreamTag.Tag.From.Name))
+			client, err := clients.NewImageClient()
+			_, err = client.ImageStreamTags(refNamespace).Get(imageStreamTag.Tag.From.Name, metav1.GetOptions{})
+			if err == nil {
+				p.Log.Info("[istag-restore] reference tag found in cluster")
+			} else {
+				p.Log.Info("[istag-restore] reference tag not found in cluster, adding to restore")
+				addImageStream := false
+				nameSplit := strings.SplitN(imageStreamTag.Tag.From.Name, ":", 2)
+				if len(nameSplit) == 2 {
```
Comment (dymurray) [inline velero-plugins/imagestreamtag/restore.go:76]: If `len(nameSplit) != 2` do we need to error out?

**Review by dymurray — APPROVED** [review]
Just a couple questions but nothing that should hold this up. Feel free to change if you think it's needed

File: velero-plugins/common/restore.go (hunk 1 above)
Comment (sseago) [reply velero-plugins/common/restore.go]: Oops. Yeah, that's a typo.

File: velero-plugins/imagestream/backup.go (hunk 2 above)
Comment (sseago) [reply velero-plugins/imagestream/backup.go:33]: I got annoyed having to sort through the whole log file to find the log statement from my plugin. This lets you grep for that string to find all logs from the plugin in question.

File: velero-plugins/imagestreamtag/restore.go (hunk 3 above)
Comment (sseago) [reply velero-plugins/imagestreamtag/restore.go:76]: Well, in theory it shouldn't happen. I didn't want to error out the whole thing based on an attempt to pull in dependent items, though. It feels like the wrong place to raise an error. If we need to error out for it here, we need to check it for all imagestreamtags. The only reason I'm checking it here is if I don't have the split, then I have no idea how I'd pull in an ImageStream here.

**Pull request merged**